rocAL/rocAL_pybind/rocal_pybind.cpp

```diff
@@ -431,26 +471,26 @@ namespace rocal{
                     py::array numpy_array;
                     switch (output_tensor->data_type())
                     {
-                    case RocalTensorOutputType::ROCAL_UINT8:
-                        numpy_array = py::array(py::buffer_info(
-                            (unsigned char *)(output_tensor->buffer()),
-                            sizeof(unsigned char),
-                            py::format_descriptor<unsigned char>::format(),
-                            output_tensor->num_of_dims(),
-                            output_tensor->dims(),
-                            output_tensor->strides()));
-                        break;
-                    case RocalTensorOutputType::ROCAL_FP32:
-                        numpy_array = py::array(py::buffer_info(
-                            (float *)(output_tensor->buffer()),
-                            sizeof(float),
-                            py::format_descriptor<float>::format(),
-                            output_tensor->num_of_dims(),
-                            output_tensor->dims(),
-                            output_tensor->strides()));
-                        break;
-                    default:
-                        throw py::type_error("Unknown rocAL data type");
+                        case RocalTensorOutputType::ROCAL_UINT8:
+                            numpy_array = py::array(py::buffer_info(
+                                (unsigned char *)(output_tensor->buffer()),
+                                sizeof(unsigned char),
+                                py::format_descriptor<unsigned char>::format(),
+                                output_tensor->num_of_dims(),
+                                output_tensor->dims(),
+                                output_tensor->strides()));
+                            break;
+                        case RocalTensorOutputType::ROCAL_FP32:
+                            numpy_array = py::array(py::buffer_info(
+                                (float *)(output_tensor->buffer()),
+                                sizeof(float),
+                                py::format_descriptor<float>::format(),
```

Comment:
Please check if it is better to use templates instead of switch case here

Reply:
copy_data_numpy_wrapper worked with templates but both this function as well as the cupy function doesn't qualify as the functions doesn't have a templated argument

Reply:
These are lambda functions